pkg/sidecar/util_posix.go

```diff
@@ -17,9 +17,56 @@ import (
 // When a new SVID is received on the updateChan, the SVID certificates
 // are stored in disk and a restart signal is sent to the proxy's process
 func (s *Sidecar) RunDaemon(ctx context.Context) error {
-	err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
-	if err != nil && !errors.Is(err, context.Canceled) {
-		return err
+	var wg sync.WaitGroup
+	ctx, cancel := context.WithCancel(ctx)
+	defer func() {
+		cancel()
+		wg.Wait()
+	}()
+
+	tasks := 0
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		tasks++
+	}
+	if s.config.JSONFilename != "" {
+		tasks++
+	}
+	if s.config.JSONFilename != "" && s.config.JwtAudience != "" {
+		tasks++
+	}
+	wg.Add(tasks)
+
+	errch := make(chan error, tasks)
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		go func() {
+			defer wg.Done()
+			errch <- workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+		}()
+	}
+	if s.config.JSONFilename != "" {
+		go func() {
+			defer wg.Done()
+			errch <- workloadapi.WatchJWTBundles(ctx, &JWTBundlesWatcher{sidecar: s}, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+		}()
+	}
+	if s.config.JSONFilename != "" && s.config.JwtAudience != "" {
+		go func() {
+			defer wg.Done()
+			s.updateJWTSVID(ctx, workloadapi.WithAddr("unix://"+s.config.AgentAddress))
+			errch <- nil
+		}()
```

Comment:
If I'm not mistaken WatchJWTBundles only can update the bundle, because JWT SVIDs are used on demand. On the issue [43](https://github.com/spiffe/spiffe-helper/issues/43) they want the same behavior as the x509 certs, where the SVID is available for use in a file. So updateJWTSVID does this, updates JWT SVID every minute (this time is an arbitrary decision) and may cause troubles for reduced TTLs. I'm open to suggestions on how to renew JWT SVIDs.

Reply:
Understood. There is no `WatchJWTContext()` function. @MarcosDY do you know why this doesn't exist?

@JU4N98 Is it possible to use the `exp` field of the JWT to know when to renew?

Reply:
Yes, i think is possible to use `exp` field. Are there problems related to renew the SVIDs on a TTL change?

Reply:
@faisal-memon  the question was if there are problems related to renew the JWT SVID if a change on the TTL occurs. I think there are no problems getting the new JWT before the actual expires. I just updated the code so the exp field is used.

Reply:
The `exp / 2` logic seems correct to me.